### PR TITLE
Improve memory efficiency

### PR DESCRIPTION
**Pull request by natoscott**

### CODE REVIEW

**Review by bjosv — COMMENTED** [review]
Interesting!
As a hint to get the tests going you probably need to increase the number of allocs, due to the table change.
Something like:

tests/ct_out_of_memory_handling.c
```
void test_alloc_failure_handling() {
 
     // Connect
     {
-        for (int i = 0; i < 128; ++i) {
+        for (int i = 0; i < 129; ++i) {
             prepare_allocation_test(cc, i);
..
-        prepare_allocation_test(cc, 128);
+        prepare_allocation_test(cc, 149);
         result = redisClusterConnect2(cc);

void test_alloc_failure_handling_async() { 
     // Connect
     {
-        for (int i = 0; i < 127; ++i) {
+        for (int i = 0; i < 128; ++i) {
             prepare_allocation_test(acc->cc, i);
 
-        prepare_allocation_test(acc->cc, 127);
+        prepare_allocation_test(acc->cc, 128);
         result = redisClusterConnect2(acc->cc);

```

File: hircluster.c [hunk 1]
```diff
@@ -1430,7 +1400,7 @@ static int cluster_update_route_by_addr(redisClusterContext *cc, const char *ip,
         for (k = (*slot_elem)->start; k <= (*slot_elem)->end; k++) {
             if (table[k] != NULL) {
                 __redisClusterSetError(cc, REDIS_ERR_OTHER,
-                                       "Diffent node hold a same slot");
+                                       "Different node holds same slot");
```
Comment (zuiderkwast) [inline hircluster.c:1403]: :100:

**Comment by zuiderkwast** [comment]
Looks great! (I didn't look at the details.)

**Review by bjosv — APPROVED** [review]
LGTM!

**Comment by natoscott** [comment]
@bjosv @zuiderkwast thanks!  Let me know if anything else remains for merge readiness.
Also, could someone with a maintainer hat register hiredis-cluster with lgtm.com for automatic checking? (redis and hiredis are there already, be great to have this project too)

**Comment by zuiderkwast** [comment]
https://lgtm.com/projects/g/Nordix/hiredis-cluster/?mode=list

**Comment by bjosv** [comment]
Thanks!

**Pull request merged**